Microsoft.Toolkit.Mvvm/Messaging/WeakReferenceMessenger.cs

```diff
@@ -421,7 +423,7 @@ public bool MoveNext()
                             this.owner.keys.Remove(node);
```

Comment:
Under the hood this remove-statement of the linked list set the node.Next property to null. Therefore the while loop will break without checking all the nodes.